app-backend/api/src/main/scala/uploads/Routes.scala

```diff
@@ -0,0 +1,85 @@
+package com.azavea.rf.api.uploads
+
+import java.util.UUID
+
+import scala.util.{Success, Failure}
+
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.model.StatusCodes
+
+import com.lonelyplanet.akka.http.extensions.{PaginationDirectives, PageRequest}
+
+import com.azavea.rf.common.{Authentication, UserErrorHandler}
+import com.azavea.rf.database.tables.Uploads
+import com.azavea.rf.database.query._
+import com.azavea.rf.database.{Database, ActionRunner}
+import com.azavea.rf.datamodel._
+import io.circe._
+import io.circe.generic.auto._
+import de.heikoseeberger.akkahttpcirce.CirceSupport._
+
+trait UploadRoutes extends Authentication
+    with UploadQueryParameterDirective
+    with PaginationDirectives
+    with UserErrorHandler
+    with ActionRunner {
+  implicit def database: Database
+
+  val uploadRoutes: Route = handleExceptions(userExceptionHandler) {
+    pathEndOrSingleSlash {
+      get { listUploads } ~
+      post { createUpload }
+    } ~
+    pathPrefix(JavaUUID) { uploadId =>
```

Comment:
Consider adding a `pathEndOrSingleSlash` directive after this. Otherwise, child routes get obscured by these top level handlers.